packages/ibm-products/src/components/Datagrid/Datagrid/addons/Filtering/hooks/useFilters.js

```diff
@@ -376,10 +378,21 @@ const useFilters = ({
             return null;
           })
           .filter(Boolean);
+        let isKeyRequired = false;
```

Comment:
i would rewrite this in a more functional way that's less reliant on `let` mutability. something like
```
const isEqual = compareFilterItems(filteredItems);
if (isEqual) { ... }
const getKey = () => isEqual ? ...
```

Reply:
Thanks David.
I've updated this now.